includes/wc-update-functions.php

```diff
@@ -1773,3 +1773,123 @@ function wc_update_340_last_active() {
+ * the command line or not. If `wp wc update` is used to update the database, this function
+ * copies data in a single go that is faster but uses more resources. If the databse update was
+ * triggered from the wp-admin, this function copies data in batches which is slower but uses
+ * few resources and thus is less likely to fail on smaller servers.
+ *
+ * @param WC_Background_Updater|false $updater Background updater instance or false if function is called from `wp wc update` WP-CLI command.
+ */
+function wc_update_350_order_customer_id( $updater = false ) {
+	global $wpdb;
+
+	$post_types              = (array) apply_filters( 'woocommerce_update_350_order_customer_id_post_types', array( 'shop_order' ) );
+	$post_types_placeholders = implode( ', ', array_fill( 0, count( $post_types ), '%s' ) );
+
+	if ( defined( 'WP_CLI' ) && WP_CLI ) {
+		// If running the update from the command-line, copy data in a single go which is faster but uses more resources.
+		$wpdb->query(
+			'CREATE TEMPORARY TABLE customers_map (post_id BIGINT(20), customer_id BIGINT(20), PRIMARY KEY(post_id))'
+		);
+
+		$wpdb->query(
+			"INSERT IGNORE INTO customers_map (SELECT post_id, meta_value FROM wp_postmeta WHERE meta_key = '_customer_user')"
+		);
+
+		$wpdb->query( 'SET sql_safe_updates=1' );
+
+		$wpdb->query(
+			$wpdb->prepare(
+				"UPDATE wp_posts JOIN customers_map ON wp_posts.ID = customers_map.post_id SET wp_posts.post_author = customers_map.customer_id WHERE post_type IN ({$post_types_placeholders})",  // phpcs:ignore WordPress.WP.PreparedSQL.NotPrepared
+				$post_types
+			)
+		);
+
+		$wpdb->update( $wpdb->posts, array( 'post_author' => 0 ), array( 'post_type' => 'shop_order_refund' ) );
+	} else {
+		// If running the update from the wp-admin, copy data in batches being careful not to use more memory than allowed.
+		$admin_orders_sql = '';
+		$admin_orders     = get_transient( 'wc_update_350_admin_orders' );
+
+		if ( false === $admin_orders ) {
+			// Get the list of orders that we don't want to change as they belong to user ID 1.
+			$admin_orders = $wpdb->get_col(
+				$wpdb->prepare(
+					"SELECT ID FROM wp_posts p
+					INNER JOIN wp_postmeta pm ON p.ID = pm.post_id
+					WHERE post_type IN ({$post_types_placeholders}) AND meta_key = '_customer_user' AND meta_value = 1", // phpcs:ignore WordPress.WP.PreparedSQL.NotPrepared
+					$post_types
+				)
+			);
+
+			// Cache the list of orders placed by the user ID 1 as to large stores this query can be slow.
+			set_transient( 'wc_update_350_admin_orders', $admin_orders, DAY_IN_SECONDS );
+		}
+
+		if ( ! empty( $admin_orders ) ) {
+			$admin_orders_sql .= ' AND ID NOT IN (' . implode( ', ', $admin_orders ) . ') ';
+		}
+
+		// Query to get a batch of orders IDs to change.
+		$query = $wpdb->prepare(
+			// phpcs:disable WordPress.WP.PreparedSQL.NotPrepared
+			"SELECT ID FROM {$wpdb->posts}
+	        WHERE post_author = 1 AND post_type IN ({$post_types_placeholders}) $admin_orders_sql
```

Comment:
indenting is odd here.

Reply:
Fixed in 397e40f1659e38627be916c6f0114b33d0c2cf2c